code/game/objects/effects/roaches.dm

```diff
@@ -0,0 +1,76 @@
+//generic procs copied from obj/effect/alien
+/obj/effect/roach
+	name = "roach effect"
+	desc = "A cockroach effect."
+	icon = 'icons/effects/effects.dmi'
+	anchored = 1
+	density = 0
+	var/health = 5
+
+/obj/effect/roach/attackby(var/obj/item/I, var/mob/user)
+	if(I.attack_verb.len)
```

Comment:
Could've condensed 11 to 14 into just `visible_message(SPAN_WARNING("\The [src] has been [I.attack_verb.len?pick(I.attack_verb):"attacked"] with \the [I][user?" by \the [user].":"."]"`

Also, wrong tense, should be has not have

Reply:
Yes sorry... I copy pasted the `attackby` of `spiders.dm` without checking the tense that was used.

I will make a small PR to correct that both in `roaches.dm` and `spiders.dm`.